src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let sep;
+
+    // in case the passed separator is null
+    if (separator === null) {
+      sep = 'null';
+    } else {
+      sep = separator;
+    }
```

Comment:
?:

Reply:
```js
test(`Should work with null`, () => { // here is the test 
  expect(source.join2(null))
    .toBe('0null1null2null3');
});

Reply:
i guess with this `?:` Misha meant the ternary
```suggestion
    separator === null
      ? sep = 'null';
      : sep = separator;
```

Reply:
🤦‍♂️ ..